beacon/validator/src/main/java/tech/pegasys/teku/validator/coordinator/performance/DefaultPerformanceTracker.java

```diff
@@ -175,23 +164,18 @@ public void onSlot(UInt64 slot) {
     }
   }
 
-  private BlockPerformance getBlockPerformanceForEpochs(
-      UInt64 startEpochInclusive, UInt64 endEpochExclusive) {
-    int numberOfBlockProductionAttempts =
-        blockProductionAttemptsByEpoch
-            .subMap(startEpochInclusive, true, endEpochExclusive, false)
-            .values()
-            .stream()
-            .mapToInt(AtomicInteger::get)
-            .sum();
-    List<SlotAndBlockRoot> producedBlocks =
-        producedBlocksByEpoch
-            .subMap(startEpochInclusive, true, endEpochExclusive, false)
-            .values()
-            .stream()
-            .flatMap(Collection::stream)
-            .collect(Collectors.toList());
+  private BlockPerformance getBlockPerformanceForEpoch(UInt64 currentEpoch) {
+    int numberOfBlockProductionAttempts = 0;
+    AtomicInteger blocksAtEpoch = blockProductionAttemptsByEpoch.get(currentEpoch);
+    if (blocksAtEpoch != null) {
+      numberOfBlockProductionAttempts = blocksAtEpoch.get();
+    }
 
+    List<SlotAndBlockRoot> producedBlocks = new ArrayList<>();
+    if (producedBlocksByEpoch.get(currentEpoch) != null) {
+      producedBlocks =
+          producedBlocksByEpoch.get(currentEpoch).stream().collect(Collectors.toList());
```

Comment:
I don't think we need to convert this to a `List` - can just make `producedBlocks` a `Collection` since the only thing we ever do with it is iterate and get the size.